Lib/test/test_pyrepl/test_unix_console.py

```diff
@@ -317,6 +317,24 @@ def test_restore_with_invalid_environ_on_macos(self, _os_write):
             console.prepare()  # needed to call restore()
             console.restore()  # this should succeed
 
+    def test_restore_in_thread(self, _os_write):
+        # for gh-139391
+        import threading
+        console = unix_console([])
+        console.old_sigwinch = signal.SIG_DFL
+        exception_caught = []
+        def thread_target():
+            try:
+                console.restore()
+            except Exception as e:
+                exception_caught.append(e)
+        thread = threading.Thread(target=thread_target)
+        thread.start()
+        thread.join()
```

Comment:
```py
    @threading_helper.reap_threads
    @threading_helper.requires_working_threading()
    def test_restore_in_thread(self, _os_write):
        # gh-139391: ensure that console.restore() silently suppresses
        # exceptions when calling signal.signal() from a non-main thread.
        console = unix_console([])
        console.old_sigwinch = signal.SIG_DFL
        thread = threading.Thread(target=console.restore)
        thread.start()
        thread.join()  # this should not raise
```

and import `threading` and `threading_helper` from `test.support` (if needed) at the top of the file.

Reply:
copy that